src/category_theory/adjunction/lifting.lean

```diff
@@ -0,0 +1,274 @@
+
+## References
+* https://ncatlab.org/nlab/show/adjoint+triangle+theorem
+* https://ncatlab.org/nlab/show/adjoint+lifting+theorem
+* Adjoint Lifting Theorems for Categories of Algebras (PT Johnstone, 1975)
+* A unified approach to the lifting of adjoints (AJ Power, 1988)
+-/
+
+namespace category_theory
+
+open category limits
+
+universes v₁ v₂ v₃ v₄ u₁ u₂ u₃ u₄
+
+variables {A : Type u₁} {B : Type u₂} {C : Type u₃}
+variables [category.{v₁} A] [category.{v₂} B] [category.{v₃} C]
+
+-- Hide implementation details in this namespace
+namespace lift_adjoint
+
+variables {U : B ⥤ C} {F : C ⥤ B} (R : A ⥤ B) (F' : C ⥤ A)
+variables (adj₁ : F ⊣ U) (adj₂ : F' ⊣ R ⋙ U)
+
+/--
+To show that `ε_X` is a coequalizer for `(FUε_X, ε_FUX)`, it suffices to assume it's always a
+coequalizer of something (i.e. a regular epi).
+-/
+def counit_coequalises [∀ (X : B), regular_epi (adj₁.counit.app X)] (X : B) :
+  is_colimit (cofork.of_π (adj₁.counit.app X) (adj₁.counit_naturality _)) :=
+cofork.is_colimit.mk' _ $ λ s,
+begin
+  refine ⟨(regular_epi.desc' (adj₁.counit.app X) s.π _).1, _, _⟩,
+  { rw ← cancel_epi (adj₁.counit.app (regular_epi.W (adj₁.counit.app X))),
+    rw ← adj₁.counit_naturality_assoc,
+    dsimp only [functor.comp_obj],
+    rw [← s.condition, ← F.map_comp_assoc, ← U.map_comp, regular_epi.w, U.map_comp,
+        F.map_comp_assoc, s.condition, ← adj₁.counit_naturality_assoc] },
+  { apply (regular_epi.desc' (adj₁.counit.app X) s.π _).2 },
+  { intros m hm,
+    rw ← cancel_epi (adj₁.counit.app X),
+    apply hm.trans (regular_epi.desc' (adj₁.counit.app X) s.π _).2.symm }
+end
+
+include adj₁ adj₂
+
+/--
+(Implementation)
+To construct the left adjoint, we use the coequalizer of `F' U ε_Y` with the composite
+
+`F' U F U X ⟶ F' U F U R F U' X ⟶ F' U R F' U X ⟶ F' U X`
+
+where the first morphism is `F' U F ι_UX`, the second is `F' U ε_RF'UX`, and the third is `δ_F'UX`.
+We will show that this coequalizer exists and that it forms the object map for a left adjoint to
+`R`.
+-/
+def other_map (X) : F'.obj (U.obj (F.obj (U.obj X))) ⟶ F'.obj (U.obj X) :=
+F'.map (U.map (F.map (adj₂.unit.app _) ≫ adj₁.counit.app _)) ≫ adj₂.counit.app _
+
+/--
+`(F'Uε_X, other_map X)` is a reflexive pair: in particular if `A` has reflexive coequalizers then
+it has a coequalizer.
+-/
+instance (X : B) :
+  is_reflexive_pair (F'.map (U.map (adj₁.counit.app X))) (other_map _ _ adj₁ adj₂ X) :=
+is_reflexive_pair.mk'
+  (F'.map (adj₁.unit.app (U.obj X)))
+  (by {rw [← F'.map_comp, adj₁.right_triangle_components], apply F'.map_id })
+  begin
+    dsimp [other_map],
+    rw [← F'.map_comp_assoc, U.map_comp, adj₁.unit_naturality_assoc, adj₁.right_triangle_components,
+        comp_id, adj₂.left_triangle_components]
+  end
+
+variables [has_reflexive_coequalizers A]
+
+/--
+Construct the object part of the desired left adjoint as the coequalizer of `F'Uε_Y` with
+`other_map`.
+-/
+noncomputable def L_obj (Y : B) : A :=
```

Comment:
Is this the best name? Is this an implementation detail?

Reply:
Everything other than the last four defs in the file are basically implementation details - I don't mind changing the name though. It's the object function for the functor `construct_left_adjoint` below so I could just call it `construct_left_adjoint_obj` - does that sound okay to you?

Reply:
yup, sounds good. please also look at the other names

Reply:
I changed `L_obj` and `L_equiv`, is there anything else you'd like me to change?